### PR TITLE
17692 Add additional corp types to entity type mapping

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#17692

*Description of changes:*

- Add additional corp types to entity type mapping, and convert to entity type as needed.
- Basically, every `legalType` passed in the entities search query string in[ `business-lookup-services.ts`](https://github.com/bcgov/namerequest/blob/main/src/services/business-lookup-services.ts#L40) needs to be a [valid entity type in namex](https://github.com/bcgov/namex/blob/main/api/namex/constants/__init__.py#L81). When it is not, we need to convert the corp type to a valid entity type before posting it to namex. This should also fix bcgov/entity#17780 and bcgov/entity#17791.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

File: src/mixins/common-mixin.ts
Comment (leodube-aot) [inline src/mixins/common-mixin.ts:93]: Couple questions here. Any change I do I'll be sure to reflect it in the code block (`entityTypeToCorpType`) above as well.
1. Are `CorpTypeCd.CONTINUE_IN` and `CorpTypeCd.LIMITED_CO` legal codes that we need to be handling in NameRequest? If yes what would be their respective `EntityTypes` codes? If not I will remove them from the search query string in [`business-lookup-services.ts`](https://github.com/bcgov/namerequest/blob/main/src/services/business-lookup-services.ts#L40)
2. What exactly is `CorpTypeCd.EXTRA_PRO_A`? I took a guess at what it's related `EntityTypes` codes would be but I may be incorrect.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

File: package.json
Comment (severinbeauvais) [inline package.json:3]: Please rebase. This version number is out of date.

File: src/components/new-request/search-components/business-lookup-fetch.vue
Comment (severinbeauvais) [inline src/components/new-request/search-components/business-lookup-fetch.vue:135]: Should you have conditional chaining and fallback to null here, same as below?

File: src/components/new-request/search-components/jurisdiction.vue
Comment (severinbeauvais) [inline src/components/new-request/search-components/jurisdiction.vue:64]: Here, too?

File: src/mixins/common-mixin.ts
Comment (severinbeauvais) [reply src/mixins/common-mixin.ts:93]: Does this help?

https://github.com/bcgov/namex/blob/068d6b6f3d7eef914b93a412424ed747ddd5846f/api/namex/services/lookup/name_request_filing_actions.py#L51

Comment (leodube-aot) [reply src/mixins/common-mixin.ts:93]: Yes it did! As for `CorpTypeCd.CONTINUE_IN`, I don't think we use it anywhere so I will be removing it.

File: src/components/new-request/search-components/business-lookup-fetch.vue
Comment (leodube-aot) [reply src/components/new-request/search-components/business-lookup-fetch.vue:135]: Yes it should be, I will update. Looking over it again this appears to be the only place it is needed. It actually isn't needed below (or in jurisdiction.vue) since they are wrapped in with `if (this.getSearchBusiness)`.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

File: src/mixins/common-mixin.ts
Comment (severinbeauvais) [reply src/mixins/common-mixin.ts:91]: 1. EntityTypes.LLC is already mapped to CorpTypeCd.LIMITED_CO (in enums/entity-types.ts).

2. I think you're right that CorpTypeCd.CONTINUE_IN (C) is not used... per this: [name request types.txt](https://github.com/bcgov/namerequest/files/12686743/name.request.types.txt)

Comment (leodube-aot) [reply src/mixins/common-mixin.ts:91]: For 1. I think I have the proper entity codes now for LIMITED_CO and EXTRA_PRO_A. For LIMITED_CO I can't use LLC since it is not one of the [entity types in namex](https://github.com/bcgov/namex/blob/main/api/namex/constants/__init__.py#L81), but namex uses RLC for Xpro Limited Liability Companies.

For 2. I'll await Mihai's reply in the team's thread.

Comment (severinbeauvais) [reply src/mixins/common-mixin.ts:91]: What did you use for LIMITED_CO?

Did you confirm that EXTRA_PRO_A is an Xpro Limited Liability Company (RLC)?

Comment (leodube-aot) [reply src/mixins/common-mixin.ts:91]: For `LIMITED_CO` I used `EntityTypes.RLC`. In the raw_nr_to_action mapping ([L62-L65](https://github.com/bcgov/namex/blob/068d6b6f3d7eef914b93a412424ed747ddd5846f/api/namex/services/lookup/name_request_filing_actions.py#L62-L65)) you provided, `LIMITED_CO` (LLC) maps to 4 different NR type codes (LC, CLC, RLC and AL, depending on the action), but only RLC (Xpro Limited Liability Company) is used in namex `EntityTypes`. It seems like we allow it to be used for restorations and name changes in [request-action-mapping.ts](https://github.com/bcgov/namerequest/blob/main/src/list-data/request-action-mapping.ts#L62).

`EXTRA_PRO_A` also maps to 4 different codes in raw_nr_to_action ([L101-L104](https://github.com/bcgov/namex/blob/068d6b6f3d7eef914b93a412424ed747ddd5846f/api/namex/services/lookup/name_request_filing_actions.py#L101-L104)): UA, XUL, XCUL, and XRUL, but only XUL (Xpro Unlimited Liability Company) is used in namex `EntityTypes`.

Comment (severinbeauvais) [reply src/mixins/common-mixin.ts:91]: I agree with you. OK by me to resolve this conversation.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-704-rqyba5nq.web.app

File: src/components/new-request/search-components/business-lookup-fetch.vue
Comment (severinbeauvais) [inline src/components/new-request/search-components/business-lookup-fetch.vue:130]: Because conditional chaining may return undefined, and an variable with a value of undefined is not reactive, provide a fallback value of null. (I think corpTypeToEntityType handles it but it's better to not depend on that.)

File: src/components/new-request/search-components/business-lookup-fetch.vue
Comment (severinbeauvais) [inline src/components/new-request/search-components/business-lookup-fetch.vue:145]: Same comment as on line 130.

Comment (severinbeauvais) [reply src/components/new-request/search-components/business-lookup-fetch.vue:145]: Or maybe not, since you didn't use conditional chaining here?

File: src/mixins/common-mixin.ts
Comment (severinbeauvais) [inline src/mixins/common-mixin.ts:73]: Great! But please sort alphabetically to make it easier for a future dev to find things.

File: src/components/new-request/search-components/business-lookup-fetch.vue
Comment (leodube-aot) [reply src/components/new-request/search-components/business-lookup-fetch.vue:145]: I don't think so in this case since getSearchBusiness will not be null here, and legalType will default to null if unassigned. A fallback will be needed for your comment on line 130 however.

**Review by severinbeauvais — APPROVED** [review]
LGTM. Please wait for another review or 2 before merging.

**Comment by leodube-aot** [comment]
@ketaki-deodhar @JazzarKarim @vysakh-menon If anyone has some time today can they review this PR please?

**Review by JazzarKarim — APPROVED** [review]
Looks great Leo! 👍 Sorry for jumping in so late to review.

**Review by ketaki-deodhar — APPROVED** [review]
Looks good!

**Pull request merged**
